new-runtimepoc/Pester.Runtime.psm1

```diff
@@ -597,9 +597,9 @@ function Invoke-TestItem {
                 -NoNewScope
 
             $Test.FrameworkData.Runtime.ExecutionStep = 'Finished'
-            $Test.Passed = $result.Success
+            $Test.ErrorRecord += ($result.ErrorRecord | Where-Object { $_.FullyQualifiedErrorId -notlike '*PesterAssertionFailed*' })
```

Comment:
Why is this filter needed? Any error in the test should mean that the test failed, not just errors from assertions. Also don't use cmdlets :)

Reply:
It is actually doing the opposite :). It is adding the errors that are not from assertions, the assertions ones are already captured as part of the `Invoke-Assertion` function in Should.ps1

I will move away from Where-Object to `foreach`

Reply:
Okay. Would that change the order of errors, though? Why are we capturing the errors in two different collections? Would this work with -ErrorAction Stop? It seems a bit overcomplicated. But we can refine later I guess.

Reply:
As far as I know, there were no order of errors before, in version 4.x you only get one *ErrorRecord* per It, so it is a breaking change anyways. This works with `-ErrorAction Stop`, but in that case you may not get a full list of errors, only the errors up to the point of the assertion with ErrorAction specified. The new order, first all assertion errors and then whatever error happened to be captured during execution

Any suggestions about this?